corehq/apps/accounting/models.py

```diff
@@ -889,6 +890,38 @@ def save(self, *args, **kwargs):
         super(SoftwarePlanVersion, self).save(*args, **kwargs)
         SoftwarePlan.get_version.clear(self.plan)
 
+    @classmethod
+    def get_most_recent_version(cls, edition=None, visibility=None):
+        # Start with all objects
+        plan_versions_query = cls.objects.all()
+
+        # Apply filters conditionally
+        if edition:
```

Comment:
this edition and visibility filtering is repeated in quite a few places and also starting on line 221. Is it possible to create a staticmethod that's something like `filter_version_query(query, edition, visibility)` to improve readability? Thanks!

Reply:
Refactored in: d20ad3dece3d47d86af2ec59ae51cb5c7eb64491
Hi Biyeun, line 221 is something unrelated to the filtering. Are you referring to something else?